the-guild-smart-contracts/src/TheGuildContributionToken.sol

```diff
@@ -0,0 +1,125 @@
+    
+    /// @notice Event emitted when tokens are minted to multiple recipients
+    event BatchMint(address indexed owner, uint256 totalAmount, uint256 recipientCount);
+    
+    /// @notice Event emitted when tokens are minted to a single recipient
+    event Mint(address indexed to, uint256 amount);
+    
+    /// @notice Event emitted when tokens are minted with a reason (e.g., GitHub ticket reference)
+    event ContributionTokenMinted(address indexed recipient, uint256 amount, bytes32 indexed reason);
+
+    /// @dev Constructor sets the token name, symbol, and initial owner
+    /// @param initialOwner The address that will own the contract
+    constructor(address initialOwner) 
+        ERC20("TheGuild Contribution Token", "TGC") 
+        Ownable(initialOwner) 
+    {
+        // Contract starts with zero supply - all tokens must be explicitly minted
+    }
+
+    /// @notice Mint tokens to a single recipient
+    /// @dev Only the owner can mint tokens
+    /// @param to The address to receive the tokens
+    /// @param amount The amount of tokens to mint (in wei, including decimals)
+    function mint(address to, uint256 amount) external onlyOwner {
+        require(to != address(0), "TGC: cannot mint to zero address");
+        require(amount > 0, "TGC: amount must be greater than zero");
+        require(totalSupply() + amount <= MAX_SUPPLY, "TGC: would exceed max supply");
+        
+        _mint(to, amount);
+        emit Mint(to, amount);
+    }
+
+    /// @notice Mint tokens to a recipient with a reason (e.g., GitHub ticket reference)
+    /// @dev Only the owner can mint tokens. Use this for tracking specific contributions.
+    /// @param to The address to receive the tokens
+    /// @param amount The amount of tokens to mint (in wei, including decimals)
+    /// @param reason The reason for minting (e.g., GitHub issue hash, ticket ID)
+    function mintWithReason(address to, uint256 amount, bytes32 reason) external onlyOwner {
+        require(to != address(0), "TGC: cannot mint to zero address");
+        require(amount > 0, "TGC: amount must be greater than zero");
+        require(reason != bytes32(0), "TGC: reason cannot be empty");
+        require(totalSupply() + amount <= MAX_SUPPLY, "TGC: would exceed max supply");
+        
+        _mint(to, amount);
+        emit ContributionTokenMinted(to, amount, reason);
+    }
+
+    /// @notice Mint tokens to multiple recipients
+    /// @dev Only the owner can mint tokens. Gas-efficient batch minting.
+    /// @param recipients Array of addresses to receive tokens
+    /// @param amounts Array of amounts to mint to each recipient
+    function batchMint(address[] calldata recipients, uint256[] calldata amounts) external onlyOwner {
+        require(recipients.length == amounts.length, "TGC: arrays length mismatch");
+        require(recipients.length > 0, "TGC: empty arrays");
+        
+        uint256 totalAmount = 0;
```

Comment:
totalAmount is not very useful imo.
Instead, please include distributionId as per described in the specs of the parent ticket: batchMint uses a distributionId as input and checks that distributionId is not in a mapping of already distributed distributionIds